daemon/k8s_watcher.go

```diff
@@ -1421,3 +1457,49 @@ func (d *Daemon) updateCiliumNetworkPolicyV2(ciliumV2Store cache.Store,
 	d.deleteCiliumNetworkPolicyV2(oldRuleCpy)
 	d.addCiliumNetworkPolicyV2(ciliumV2Store, newRuleCpy)
 }
+
+func (d *Daemon) updateK8sPodV1(oldK8sPod, newK8sPod *v1.Pod) {
+	if oldK8sPod == nil || newK8sPod == nil {
+		return
+	}
+
+	// We only care about label updates
+	oldPodLabels := oldK8sPod.GetLabels()
+	newPodLabels := newK8sPod.GetLabels()
+	if reflect.DeepEqual(oldPodLabels, newPodLabels) {
```

Comment:
I think this is fine but we might want to write a custom function to compare as this will be hit often.